### PR TITLE
Add Quickstart Guide to docs.base.org/docs

### PR DESCRIPTION
**Pull request by ericbrown99**
**What changed? Why?**
Added a Quickstart: Deploy on Base guide to the `Building on Base` section of the docs. 

The intention of this guide is to stay as high level as possible while still providing enough information for new developers to deploy contracts to Base. It uses the Foundry framework to support this process. 

**Notes to reviewers**
Please review for correctness and consistency of language. Please follow the tutorial and ensure you are able to replicate the results by exactly following the steps outlined in the guide to ensure everyone will have success. 

**How has it been tested?**
I tested this by spinning up a fresh directory and following the steps exactly to ensure I was able to deploy and interact with contracts from my local machine.

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/2</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>1</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>2</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

**Comment by ericbrown99** [comment]
cc @wbnns & @hughescoin

File: apps/base-docs/docs/building-with-base/quickstart.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+# Quick Start: Deploy on Base
+
+Welcome to the Base deployment quickstart guide! This comprehensive walkthrough will help you set up your environment and deploy smart contracts on Base. Whether you're a seasoned developer or just starting out, this guide has got you covered.
+
+## What You'll Achieve
+
+By the end of this quickstart, you'll be able to:
+
+- Set up your develompent environment to deploy on Base
```
Comment (wbnns) [inline apps/base-docs/docs/building-with-base/quickstart.md:28]: s/develompent/development/

File: apps/base-docs/docs/building-with-base/quickstart.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+Your private key is stored in `~/.foundry/keystores` which is not tracked by git.
+
+:::warning
+Never share or commit your private key. Always keep it secure and handle with care.
+:::
+
+## Deploy Your Contracts
+
+Now that your enviornment is set up, let's deploy your contracts to Base Sepolia.
```
Comment (wbnns) [inline apps/base-docs/docs/building-with-base/quickstart.md:114]: s/enviornment/environment/

**Review by wbnns — CHANGES_REQUESTED** [review]
@ericm-cb 

Heya! Thanks for putting this together! Left two comments in-line w/ small nits but otherwise LGTM! Can you please merge in the latest changes from `master`?

Also, a question -- I noticed what looked like maybe a couple of artifacts in these commit titles:
c5fdd91a1212fb42616f616deb46315ce6bde2b3
3f48d4c2fbd8d870f9ef32ee01c330aefdc4be9d

 I didn't see any translations in this PR. PLMK if I should be looking somewhere else for the correlation here. 👍

**Comment by ericbrown99** [comment]
> Also, a question -- I noticed what looked like maybe a couple of artifacts in these commit titles: [c5fdd91](https://github.com/base-org/web/commit/c5fdd91a1212fb42616f616deb46315ce6bde2b3) [3f48d4c](https://github.com/base-org/web/commit/3f48d4c2fbd8d870f9ef32ee01c330aefdc4be9d)
> 
> I didn't see any translations in this PR. PLMK if I should be looking somewhere else for the correlation here. 👍

Good catch, fixed the artifacts 👍

**Comment by ericbrown99** [comment]
@wbnns & @hughescoin 

Fixed the open items above and merged in the latest from master, thanks! Mind taking a last quick look and then LGTM 🚀

**Review by wbnns — CHANGES_REQUESTED** [review]
@ericbrown99 

Heya, looks like there is an artifact in `yarn.lock` which is causing the build to fail

**Comment by cb-heimdall** [comment]
Review Error for wbnns @ 2024-09-26 20:21:58 UTC
User failed mfa authentication, public email is not set on your github profile. see [go/mfa-help](http://go/mfa-help)

**Review by hughescoin — APPROVED** [review]

**Comment by cb-heimdall** [comment]
Review Error for hughescoin @ 2024-09-26 21:31:46 UTC
User failed mfa authentication, see [go/mfa-help](http://go/mfa-help)

**Review by hughescoin — APPROVED** [review]

**Pull request merged**